### PR TITLE
Changes for pagination on viewprofiles tab

### PR DESCRIPTION
**Pull request by nikhilaghai**
Changes have been doen to implement pagination of profile list on the  view profiles tab

### CODE REVIEW

File: src/components/ProfilesTable.js [hunk 1]
```diff
@@ -45,6 +50,11 @@ class CustomTable extends React.Component {
 
     this.setState({ validated: true });
   };
+  handlePageChange = pageNumber =>{
+    this.setState({
+      activePage: pageNumber-1},() => { console.log('state', this.state);
```
Comment (CGIBIFE) [inline src/components/ProfilesTable.js]: Please remove the console.

File: src/reducers/profileReducer.js [hunk 2]
```diff
@@ -15,9 +17,20 @@ const initialState = {
...
 
          }
-    }
+    },
+    paginatedProfiles:[],
+    startIndex:0
 };
 
+const getAllButRejectedProfiles = profiles => {
+    return profiles.filter(profile => profile.status != "Rejected");
```
Comment (CGIBIFE) [inline src/reducers/profileReducer.js]: Please strict check with !== (Not mandatory but best practice)

File: src/reducers/profileReducer.js [hunk 3]
```diff
@@ -15,9 +17,20 @@ const initialState = {
...
+}
+
+const applyPagination = (activePage, profiles) => {
+    return getAllButRejectedProfiles(profiles).slice((activePage-1) * constants.pageSize,
+       (constants.pageSize * activePage))
+}
+
 const activeProfile = ["Application Received", "Interview Scheduled", "Offer Made", "On Hold"]
```
Comment (CGIBIFE) [inline src/reducers/profileReducer.js]: I think we repeat this array many times so it is the right time to move this is into constant

File: src/reducers/profileReducer.js [hunk 4]
```diff
@@ -26,9 +39,10 @@ function profileReducer(state = initialState, action) {
...
             ...state,
             profiles: action.data,
+            paginatedProfiles : applyPagination(1,action.data),
             activeProfiles: action.data.filter(profile =>
                 activeProfile.includes(profile.status)),
-            allButRejectedProfiles: action.data.filter(profile => profile.status != "Rejected"),
+            allButRejectedProfiles: getAllButRejectedProfiles(action.data),
             stats:{
               pieChart:{                
                 offerProfilesCount: action.data.filter(profile => profile.status == "Offer Made").length,
```
Comment (CGIBIFE) [inline src/reducers/profileReducer.js]: Please use Constant for the "offer made" string

File: src/components/ProfilesTable.js (hunk 1 above)
Comment (nikhilaghai) [reply src/components/ProfilesTable.js]: This code is removed as the logic is moved to another component

File: src/reducers/profileReducer.js (hunk 2 above)
Comment (nikhilaghai) [reply src/reducers/profileReducer.js]: Fixed

File: src/reducers/profileReducer.js (hunk 3 above)
Comment (nikhilaghai) [reply src/reducers/profileReducer.js]: This array has been moved to the constants file

File: src/reducers/profileReducer.js (hunk 4 above)
Comment (nikhilaghai) [reply src/reducers/profileReducer.js]: Constant created

**Review by CGIBIFE — APPROVED** [review]

**Pull request merged**